blocks/test/full-content.js

```diff
@@ -93,110 +81,51 @@ describe( 'full post content fixture', () => {
-				} else {
-					throw new Error(
-						'Missing fixture file: ' + f + '.parsed.json'
-					);
-				}
-			}
+			beforeAll( () => {
+				content = readFixtureFile( f + '.html' );
+			} );
 
-			const parserOutputExpected = JSON.parse( parserOutputExpectedString );
-			try {
-				expect(
-					parserOutputActual
-				).toEqual( parserOutputExpected );
-			} catch ( err ) {
-				throw new Error( format(
-					'File \'%s.parsed.json\' does not match expected value:\n\n%s',
-					f,
-					err.message
-				) );
-			}
+			it( `has non empty fixture file ${ f }.html`, () => {
+				expect( content ).toBeTruthy();
+			} );
 
-			const blocksActual = parse( content );
-			const blocksActualNormalized = normalizeParsedBlocks( blocksActual );
-			let blocksExpectedString = readFixtureFile( f + '.json' );
-
-			if ( ! blocksExpectedString ) {
-				if ( process.env.GENERATE_MISSING_FIXTURES ) {
-					blocksExpectedString = JSON.stringify(
-						blocksActualNormalized,
-						null,
-						4
-					) + '\n';
-					writeFixtureFile( f + '.json', blocksExpectedString );
-				} else {
-					throw new Error(
-						'Missing fixture file: ' + f + '.json'
-					);
-				}
-			}
+			it( 'gets parsed properly', () => {
+				const parserOutput = grammarParse( content );
 
-			const blocksExpected = JSON.parse( blocksExpectedString );
-			try {
-				expect(
-					blocksActualNormalized
-				).toEqual( blocksExpected );
-			} catch ( err ) {
-				throw new Error( format(
-					'File \'%s.json\' does not match expected value:\n\n%s',
-					f,
-					err.message
-				) );
-			}
+				expect( parserOutput ).toMatchSnapshot();
+			} );
 
-			const serializedActual = serialize( blocksActual );
-			let serializedExpected = readFixtureFile( f + '.serialized.html' );
-
-			if ( ! serializedExpected ) {
-				if ( process.env.GENERATE_MISSING_FIXTURES ) {
-					serializedExpected = serializedActual;
-					writeFixtureFile( f + '.serialized.html', serializedExpected );
-				} else {
-					throw new Error(
-						'Missing fixture file: ' + f + '.serialized.html'
-					);
-				}
-			}
+			it( 'gets blocks normalized properly', () => {
+				const blocks = parse( content );
+				const blocksNormalized = normalizeParsedBlocks( blocks );
 
-			try {
-				expect( serializedActual ).toEqual(
-					serializedExpected.replace( /\n$/, '' )
-				);
-			} catch ( err ) {
-				throw new Error( format(
-					'File \'%s.serialized.html\' does not match expected value:\n\n%s',
-					f,
-					err.message
-				) );
-			}
+				expect( blocksNormalized ).toMatchSnapshot();
+			} );
+
+			it( 'gets parsed and serialized properly', () => {
+				const blocks = parse( content );
+				const serialized = serialize( blocks );
+
+				expect( serialized ).toMatchSnapshot();
+			} );
 		} );
 	} );
 
 	it( 'should be present for each block', () => {
 		const errors = [];
+		const blockTypes = getBlockTypes();
+
+		expect( blockTypes ).toHaveLength( 51 );
```

Comment:
Again, this check would help us control how many block types are processed. If we don't want to keep it updated whenever block type is added/removed, we should at least verify if this number is greater than 0.